[R25.09] feat: added first version of the data trust & security KIT

matbmoser opened the pull request:
Preview: https://matbmoser.github.io/next-eclipse-tractusx.github.io/docs-kits/next/kits/data-trust-and-security-kit/adoption-view

Plan is to merge this KIT as a first version in R25.09, so we can expand its content and get sufficient feedback for it to become mature in R25.12.

R25.09 Release Check Ticket: https://github.com/eclipse-tractusx/sig-release/issues/1495

IP Check Issue Approved: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/22731

## Description

Included first version of data trust & security kit worked in: https://github.com/manourym/CX-KIT-development/tree/data-trust-and-security-kit/docs-kits/kits/data-trust-and-security-kit

I tried to merge it directly but there was many conflicts.

This PR is related to this ticket: https://github.com/eclipse-tractusx/sig-release/issues/1281

- Added Digital Product Pass Verification Success Story
- Added Software View with overview based on https://github.com/eclipse-tractusx/digital-product-pass/blob/main/dpp-verification/README.md
- Added Operation View initial structure
- Added template for rulebook
- Added examples of Verifiable Credentials
- Added rough definition of an Attestation Provider.
- Added example and explanations on how to enable SAMM Aspect Models in verifiable credentials.
- Added example on how to bring a Verifiable Credential (for PCF, Certificate Management & DPP) in a Digital Twin.

Closes eclipse-tractusx/sig-release#1281

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [X] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [X] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

matbmoser commented on docs-kits/kits/data-trust-and-security-kit/software-development-view/verifiable-credentials.md:
```diff
@@ -0,0 +1,753 @@
+
+1. **Credential Retrieval**: Obtain verifiable credentials from data providers
+2. **Issuer Validation**: Verify issuer is in trusted registry and status is active
+3. **Signature Verification**: Validate cryptographic signatures and proof integrity
```
change this

matbmoser commented:
Ok I finished the content which we talked in the meetings about. Please review so we can merge it.

marcelruland commented on docs-kits/kits/data-trust-and-security-kit/software-development-view/overview.md:
```diff
@@ -0,0 +1,308 @@
+|-----------|-------------|---------------------|-------------|
+| Digital Twin Registry | Registry for digital twin metadata and endpoints | eclipse-tractusx/sldt-digital-twin-registry | CX-0002 |
+| Eclipse Dataspace Components (EDC) Connector | Data exchange infrastructure | eclipse-tractusx/tractusx-edc | CX-0018 |
+| Wallet Service | SSI wallet for credential management | eclipse-tractusx/identity-hub | CX-0049 |
```
Not yet ready. Standard is CX-0149.

matbmoser replied:
removed all standards

marcelruland commented on docs-kits/kits/data-trust-and-security-kit/software-development-view/overview.md:
```diff
@@ -0,0 +1,308 @@
+| Digital Twin Registry | Registry for digital twin metadata and endpoints | eclipse-tractusx/sldt-digital-twin-registry | CX-0002 |
+| Eclipse Dataspace Components (EDC) Connector | Data exchange infrastructure | eclipse-tractusx/tractusx-edc | CX-0018 |
+| Wallet Service | SSI wallet for credential management | eclipse-tractusx/identity-hub | CX-0049 |
+| Operator Wallet | SSI wallet for credential issuance | eclipse-tractusx/ssi-credential-issuer | CX-0049 |
```
Needs a wallet to function, cannot standalone. Is a CSP-specific component, not needed by anyone else.

matbmoser replied:
removed standards

marcelruland commented on docs-kits/kits/data-trust-and-security-kit/software-development-view/verifiable-credentials.md:
```diff
@@ -0,0 +1,1125 @@
+
+Here's how it works in practice: Organizations use digital wallets (identified by their DIDs & optionally Business Partner Numbers) to create and manage these certificates. They sign certificates with their private keys, and you can verify them using their public keys - similar to how secure websites work with HTTPS certificates.
+
+## Types of Digital Certificates
```
These are use-case independent credential types. Please align with the SSI EG in an earlier state.

marcelruland commented on docs-kits/kits/data-trust-and-security-kit/software-development-view/api-specifications.md:
```diff
@@ -0,0 +1,608 @@
+
+The Data Trust & Security KIT provides REST APIs that make it easy to add trust and verification features to your applications. Think of these APIs as the building blocks you'll use to check whether data is trustworthy.
+
+#### Trusted List API - Finding Qualified Verifiers
```
Please align with SSI EG.

matbmoser replied:
already done

marcelruland commented on docs-kits/kits/data-trust-and-security-kit/software-development-view/api-specifications.md:
```diff
@@ -0,0 +1,608 @@
+| What You Want to Do | HTTP Method | Endpoint | What You Send |
+|---------------------|-------------|----------|---------------|
+| Get list of trusted verifiers | GET | `/issuers` | Use case type, status filter |
+| Get details about a specific verifier | GET | `/issuers/{did}` | Organization's digital ID |
```
Querying individual verifiers poses a correlation risk. Trust lists should only have `getall` endpoints.

marcelruland reviewed (changes requested):
What is the scenario we are solving here? The attestation provider role verifies the correctness of data (e.g. of a PCF calculation). So is this it? I.e. an answer to the question "How can I ensure that the data I received is sound?"?

The contents have significant overlap with ongoing work in the SSI EG (e.g. trust lists and VC types) and need to be aligned, especially since the contents are use-case-independent and on a detailed level (e.g. available trust list statuses, API definitions, etc). Please align in a significantly earlier stage in the future.

Some services don't work as indicated here. The Identity Hub will not be feature-complete before Q3 2026. The [issuer service](https://github.com/eclipse-tractusx/ssi-credential-issuer/) does not work without a wallet and is a CSP-specific service, not needed by other participants.

I've left some exemplary comments, but not at all exhaustive.

matbmoser commented:
Thinks that are missing:

- Defining the @contexts for the verifiable credentials, and hosting them in the semantic models.
- Specifying how the rulebook can be used.
- How is the verifiable credentials stored in the Digital Twin and retrieved using the EDC (for the DAC is defined, but is missing for the AAC AND AAC-SD)

SmietaJa commented on docs-kits/kits/data-trust-and-security-kit/adoption-view.md:
In my view we cannot take off responsibility of authentication data from data provider.

SmietaJa commented on docs-kits/kits/data-trust-and-security-kit/adoption-view.md:
Attestation Provider: This should serve as an additional entity capable of confirming the authenticity of data through validation and verification. However, the primary responsibility for the data must remain with the data provider."

matbmoser commented:
The proposal is to add this KIT in a "draft" status, and keep in the release, it would fit, because in 25.12 we need to work on refining all the KITs and bringing more quality to them. 

But at least the KIT would be introduced, and the work that was done would also be published (also when it does not fit 100%, since is the first version of the KIT)

matbmoser commented:
I have added the JSON Schema to JSON-LD context translator for SAMM generated schemas in the Tracutus-X SDK: https://github.com/eclipse-tractusx/tractusx-sdk/blob/main/src/tractusx_sdk/extensions/semantics/schema_to_context_translator.py

I have also refactored the credential structures and added some examples.

arnoweiss commented:
@matbmoser - do you have a deployment of the to-be state (ie somewhere in a fork)?

matbmoser commented:
> @matbmoser - do you have a deployment of the to-be state (ie somewhere in a fork)?

No I execute it in local, with npm run start, but it works.

matbmoser commented:
@nik01a5 please review.

matbmoser commented:
I have informed the stakeholders interested in this KIT, that their review is needed, making it transparent that if there is no interest that its release should be postponed to R25.12

matbmoser commented:
I have solved the problem why it was not building. Now it is better defined, there was a problem because this KIT is new, so there was broken links to the latest version. We need to point exactly for the next version.

ClosedSourcerer commented on docs-kits/kits/data-trust-and-security-kit/adoption-view.md:
````diff
@@ -0,0 +1,162 @@
+end
+```
+
+For that definition of various terms such as verification and validation within the data ecosystem, data models used to describe data, initiatives already adressing trust as well as technical solutions to verify and validate data.
````
adressing -->addressing

ClosedSourcerer commented on docs-kits/kits/data-trust-and-security-kit/adoption-view.md:
```diff
@@ -0,0 +1,162 @@
+
+This Data Certification/Verification Framework aims to create a second layer of trust over the EDC data exchanges between consumers and data providers. It enables auditors to verify specific attributes or complete aspect models for data providers and allowing consumers to retrieve and verify the "validity" of the verification done. Using a simple wallet, a Data Provider is able to certify its att ...
+
+The concept is the First Aspect Model Verification/Certification Concept in Tractus-X. It aims to provide a "lighthouse" for any other aspect model verification/certification that MUST be done in Tractus-X using SAMM Aspect Models. It provides a generic concept for Attribute Verification/Certification by external/internal auditors, and also provides a Self-Attestation option for Data Providers to ...
```
sovereighty-->sovereignty

ClosedSourcerer commented on docs-kits/kits/data-trust-and-security-kit/software-development-view/semantic-verification.md:
```diff
@@ -0,0 +1,2308 @@
+    participant Parser as sammSchemaParser
+    participant SemanticHub
+    participant VC as Verifiable Credential
+    participant IssuanceService as Issance Service
```
Issance --> Issuance

ClosedSourcerer reviewed (changes requested):
Minor Spelling Suggestions

marcelruland reviewed (dismissed):
I've only left some exemplary comments. Let's discuss the main content on Friday.

marcelruland commented on docs-kits/kits/data-trust-and-security-kit/adoption-view.md:
```diff
@@ -165,32 +127,16 @@ The Data Trust & Security KIT is built upon established industry standards and s
-| **JSON-LD** | 1.1 | Linked data format for structured information | [W3C JSON-LD](https://www.w3.org/TR/json-ld11/) |
+| Standard | Version | Reference |
+|----------|---------|-----------|
+| **Verifiable Credentials Data Model** | 1.1 & 2.0 | [W3C VC Data Model 2.0](https://www.w3.org/TR/vc-data-model-2.0/) |
```
It's fine if you stick to v2.0 in the KIT when there's a requirement for it. I'd avoid mixing two versions, it's unnecessary hassle.

ClosedSourcerer reviewed (approved):
LGTM

As a side-node:

In the adoption View: docs-kits/next/kits/data-trust-and-security-kit/adoption-view

There is `Vision & Mission`
as well as `Who & Why` in a mermaid diagram.

The font used by mermaid is `trebuchet ms`. This is not defined in the diagram itself or in the kit but must be defined more globally.

`trebuchet ms` looks completely different from fonts used elsewhere in the KIT. Especially `&` and `g` stand out like a sore thumb: 
<img width="47" height="56" alt="image" src="https://github.com/user-attachments/assets/919fa29d-7d4f-4489-afad-b26de04dac26" />
<img width="33" height="47" alt="image" src="https://github.com/user-attachments/assets/c4d11c29-b38b-497e-ac29-b9178d0d2c5e" />

Could we maybe align the fonts please?

matbmoser commented:
I have open an IP check, because there is more than 1.000 LOC: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/22731

Also when it is documentation..

matbmoser commented:
IP Checks Approved: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/22731

manourym commented on docs-kits/kits/data-trust-and-security-kit/adoption-view.md:
```diff
@@ -0,0 +1,168 @@
+
+#### Mission
+
+This KIT aims to define solutions used in Eclipse Tractus-X to generate trust in the data consumed within the network.
```
We could even mention here, that DPP and PCF are the first examples and a refined version will follow.

matbmoser commented:
@marcelruland I have added the relevant Catena-X standards, please review if I forgot something.

Added also copyright information from the companies which collaborated. Additionally I also have added some more context at the adoption view, because it was a bit empty.

It will be ready to merge now :)

hschwabe commented:
@matbmoser  Hi Mathias, thanks for all the great work and leadership. However, I would ask to remove the direct reference to Chem-X in "Governance" for now, since this draft is not formally endorsed by all Chem-X members. We can do that later.

Having said that, you can mention that there are many industry projects in Manufacturing-X which will benefit.

Text proposal: https://matbmoser.github.io/next-eclipse-tractusx.github.io/docs-kits/next/kits/data-trust-and-security-kit/adoption-view#layer-1-governance

"Attestation providers and governance bodies can be industry-specific (LINK to CX Operating Model) or cross-industry, as currently developed by the Manufacturing-X Initiative (LINK: https://www.plattform-i40.de/IP/Navigation/EN/Manufacturing-X/Manufacturing-X.html)"

matbmoser commented:
> @matbmoser Hi Mathias, thanks for all the great work and leadership. However, I would ask to remove the direct reference to Chem-X in "Governance" for now, since this draft is not formally endorsed by all Chem-X members. We can do that later.
> 
> Having said that, you can mention that there are many industry projects in Manufacturing-X which will benefit.
> 
> Text proposal: https://matbmoser.github.io/next-eclipse-tractusx.github.io/docs-kits/next/kits/data-trust-and-security-kit/adoption-view#layer-1-governance
> 
> "Attestation providers and governance bodies can be industry-specific (LINK to CX Operating Model) or cross-industry, as currently developed by the Manufacturing-X Initiative (LINK: https://www.plattform-i40.de/IP/Navigation/EN/Manufacturing-X/Manufacturing-X.html)"

Hi @hschwabe,

I have removed chem-x from the governance part, since it is also not clear how will be the future of governance bodies you are right. 

I have included your text explanation where you requested, I think it fits very well and it adds extra context explanation to this section.

Thank you for the review :) It is good to have the perspective of other -X initiatives in our KITs

matbmoser commented:
I updated the licenses regarding some feedback I received per email. If there is still some incongruence please feel free to mention here or personally to me at mathias.moser@catena-x.net

marcelruland reviewed (changes requested):
Only two comments.

matbmoser commented:
TÜV SÜD requested to be called: TÜV SÜD Auto Service GmbH

stephanbcbauer reviewed (approved):
LGTM as discussed in many meetings

matbmoser commented on docs-kits/kits/data-trust-and-security-kit/software-development-view/semantic-verification.md:
````diff
@@ -0,0 +1,3418 @@
+```
+
+### The Translation Solution
+
````
we can include that in the next kIT release.

matbmoser replied:
I have added a disclaimer.

matbmoser commented on docs-kits/kits/data-trust-and-security-kit/software-development-view/semantic-verification.md:
````diff
@@ -0,0 +1,3418 @@
+This diagram shows the current incompatibility:
+
+```mermaid
+flowchart TD
````
```suggestion
flowchart TD
      VC["Verifiable Credentials"] -- USE --> JLD["JSON-LD"]
      JLD["JSON-LD"] -- REQUIRES --> JLDC["@context"]
      SAMM["SAMM"] -- GENERATES NO --> JLDC["@context"]
      SM["TX Semantic Models"] -- ARE MODELLED IN --> SAMM["SAMM"]
      SAMM["SAMM"] -- GENERATES --> JSS["JSON SCHEMA"]
      JSS["JSON-SCHEMA"] -- CAN BE TRANSLATED TO --> JLDC["@context"]
      SAMM["SAMM"] -- CAN EXPORT SEMANTICS AS --> OWLRDF["OWL/RDF"]
      OWLRDF["OWL/RDF"] -- GENERATE --> JLDC["@context"]
```

matbmoser commented on docs-kits/kits/data-trust-and-security-kit/software-development-view/semantic-verification.md:
````diff
@@ -0,0 +1,3418 @@
+flowchart LR
+    SM["Tractus-X Data Models"] x--NOT COMPATIBLE WITHOUT TRANSLATION --x VC["Verifiable Credentials"]
+```
+
````
```suggestion
So some sort of translation is necessary. Or directly from the JSON-Schemas or using the OWL or RDF format.

```

matbmoser commented on docs-kits/kits/data-trust-and-security-kit/software-development-view/semantic-verification.md:
```diff
@@ -0,0 +1,3418 @@
+
+### The Translation Solution
+
+We solve this by automatically converting SAMM JSON schemas into JSON-LD contexts:
```
```suggestion
:::info

More information about translating SAMM into OWL and RDF and then translating to JSON-LD context is still not available in this KIT. It will be available probably in the next version.

:::

One way of solving this is by automatically converting SAMM JSON schemas into JSON-LD contexts:

```

matbmoser commented on docs-kits/kits/data-trust-and-security-kit/adoption-view.md:
```diff
@@ -0,0 +1,168 @@
+For that definition of various terms such as verification and validation within the data ecosystem, data models used to describe data, initiatives already addressing trust as well as technical solutions to verify and validate data.
+Additionally, a template for different rulebooks to gain trust in specific use case is defined.
+
+#### Customer Journey
```
lets remove it

matbmoser commented on docs-kits/kits/data-trust-and-security-kit/adoption-view.md:
```diff
@@ -0,0 +1,236 @@
+For that definition of various terms such as verification and validation within the data ecosystem, data models used to describe data, initiatives already addressing trust as well as technical solutions to verify and validate data.
+Additionally, a template for different rulebooks to gain trust in specific use case is defined.
+
+#### Customer Journey
+
```
```suggestion
```

stephanbcbauer reviewed (approved):
LGTM, as already discussed :)

stephanbcbauer commented:
> LGTM, as already discussed :)

Please have a look ⇾ markdown linting issues

matbmoser commented:
I have tried to solve the markdown linting problem, and also have added a disclaimer about the digital twin compliance, so we are transparent.
